src/test/java/com/impactupgrade/nucleus/it/MailchimpCommunicationServiceIT.java

```diff
@@ -0,0 +1,191 @@
+      if (communicationService instanceof MailchimpCommunicationService) {
+        communicationService.syncContacts(beforeBulkImport);
+      }
+    }
+
+    EnvironmentConfig.Mailchimp mailchimp = env.getConfig().mailchimp.get(0);
+    String listId = mailchimp.lists.get(0).id;
+    MailchimpClient mailchimpClient = new MailchimpClient(mailchimp, env);
+
+    assertEmailsStatus(emails, "unsubscribed", listId, mailchimpClient);
+  }
+
+  @Test
+  public void syncUnsubscribes() throws Exception {
+    List<String> unsubscribeEmails = new ArrayList<>();
+    List<String> cleanEmails = new ArrayList<>();
+    List<Object> values = new LinkedList<>();
+    values.add(List.of("Contact First Name", "Contact Last Name", "Contact Email", "Contact Email Opt In"));
+
+    for (int i = 1; i <= 5; i++) {
+      String firstname = RandomStringUtils.randomAlphabetic(8);
+      String lastname = RandomStringUtils.randomAlphabetic(8);
+      String email = (firstname + "." + lastname + "@gmail.com").toLowerCase(Locale.ROOT);
+
+      ((i % 2 == 0) ? unsubscribeEmails : cleanEmails).add(email);
+      values.add(List.of(firstname, lastname, email, "true"));
+    }
+
+    List<String> emails = new ArrayList<>();
+    emails.addAll(unsubscribeEmails);
+    emails.addAll(cleanEmails);
+
+    // Bulk import to SF
+    Instant instant = LocalDate.now().atStartOfDay(ZoneId.of("UTC")).toInstant();
+    Calendar beforeBulkImport = Calendar.getInstance();
+    beforeBulkImport.setTimeInMillis(instant.toEpochMilli());
+
+    postToBulkImport(values);
+
+    List<CrmContact> crmContacts = env.primaryCrmService().getContactsByEmails(emails);
+    assertNotNull(crmContacts);
+
+    List<String> crmContactsEmails = crmContacts.stream().map(crmContact -> crmContact.email).collect(Collectors.toList());
+    assertEquals(emails.size(), crmContactsEmails.size());
+    assertTrue(crmContactsEmails.containsAll(emails));
+
+    // Subscribe emails to MC list
+    EnvironmentConfig.Mailchimp mailchimp = env.getConfig().mailchimp.get(0);
+    String listId = mailchimp.lists.get(0).id;
+    MailchimpClient mailchimpClient = new MailchimpClient(mailchimp, env);
+
+    addEmailsToList(emails, "subscribed", listId, mailchimpClient);
+    Thread.sleep(10000);
+
+    // Update statuses to unsubscribed / cleaned
+    updateEmailsStatus(unsubscribeEmails, "unsubscribed", listId, mailchimpClient);
+    assertEmailsStatus(unsubscribeEmails, "unsubscribed", listId, mailchimpClient);
+
+    updateEmailsStatus(cleanEmails, "cleaned", listId, mailchimpClient);
+    assertEmailsStatus(cleanEmails, "cleaned", listId, mailchimpClient);
+
+    // Sync unsubscribes MC >> SF
+    for (CommunicationService communicationService : env.allCommunicationServices()) {
+      if (communicationService instanceof MailchimpCommunicationService) {
+        communicationService.syncUnsubscribes(beforeBulkImport);
+      }
+    }
+
+    crmContacts = env.primaryCrmService().getContactsByEmails(unsubscribeEmails);
```

Comment:
If the list is empty, the test won't fail since `assertTrue` is in the `for` loop. Before the loop, assert that the list isn't empty?